gm4_ziprails/data/gm4/advancements/ziprails.json

```diff
@@ -8,22 +8,43 @@
       "translate": "%1$s%3427655$s",
       "with": [
         "Wheeeeeeeee",
-        {"translate": "advancement.gm4.ziprails.title"}
+        {
+          "translate": "advancement.gm4.ziprails.title"
+        }
       ]
     },
     "description": {
       "translate": "%1$s%3427655$s",
       "with": [
         "Fly through the sky on a ziprail.",
-        {"translate": "advancement.gm4.ziprails.description"}
+        {
+          "translate": "advancement.gm4.ziprails.description"
+        }
       ],
       "color": "gray"
     }
   },
   "parent": "gm4:root",
   "criteria": {
     "ride_ziprail": {
-      "trigger": "minecraft:impossible"
+      "trigger": "minecraft:location",
+      "conditions": {
+        "player": [{
+          "condition": "minecraft:entity_properties",
+          "entity": "this",
+          "predicate": {
+            "vehicle": {
+              "type": "minecraft:minecart",
+              "nbt": "{Tags:[\"gm4_linked\"]}"
+            },
+            "player": {
+              "advancements": {
+                "gm4:ziprails": false

```

Comment:
Adding this check is useless and hurts performance. When the player already has this advancement it won't trigger it again and even start to check conditions. For players that don't have the advancement it's extra work that always gives the same result. (Also side note please avoid formatting the document using `[{`, line 32).